extensions/netty/deployment/src/main/java/io/quarkus/netty/deployment/NettyProcessor.java

```diff
@@ -115,6 +117,9 @@ NativeImageConfigBuildItem build(
                 .addRuntimeReinitializedClass("io.netty.util.internal.PlatformDependent0")
                 .addNativeImageSystemProperty("io.netty.leakDetection.level", "DISABLED");
 
+        // Also set io.netty.noUnsafe at runtime
+        systemProperties.produce(new SystemPropertyBuildItem("io.netty.noUnsafe", "true"));
```

Comment:
Good catch. Don't we need the same for `io.netty.allocator.maxOrder` though?

Reply:
No, we don't as it's already set in `limitArenaSize` (and is set there both for native and JVM).